adserver/models.py

```diff
@@ -632,14 +632,27 @@ def weighted_clicks_needed_today(self):
         Calculates clicks needed taking into account a flight's priority.
 
         For the purpose of clicks needed, 1000 impressions = 1 click (for CPM ads)
+        Takes into account value of the flight,
+        which causes higher paid and better CTR ads to be prioritized.
         """
         impressions_needed = 0
 
         # This is naive but we are counting a click as being worth 1,000 views
         impressions_needed += math.ceil(self.views_needed_today() / 1000.0)
         impressions_needed += self.clicks_needed_today()
 
-        return impressions_needed * self.priority_multiplier
+        # Cost should be a float from around .5-5, averaging 2
+        cost = float(self.cpm or self.cpc)
+
+        # CTR should be a float around .1
+        # Leading to a base value of $2 * 5 * .1 == 1.0
+        ctr_value = cost * 5.0 * float(self.ctr())
```

Comment:
So if I'm reading this right, a $4 ad with a CTR of 0.2% would be ~4x as likely to be shown. That seems reasonable.